lnd.go

```diff
@@ -119,10 +120,113 @@ func lndMain() error {
 		}
 	}
 
+	cert, err := tls.LoadX509KeyPair(cfg.TLSCertPath, cfg.TLSKeyPath)
+	if err != nil {
+		return err
+	}
+	tlsConf := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		/*
+		 * These cipher suites fit the following criteria:
+		 * - Don't use outdated algorithms like SHA-1 and 3DES
+		 * - Don't use ECB mode or other insecure symmetric methods
+		 * - Included in the TLS v1.2 suite
+		 * - Are available in the Go 1.7.6 standard library (more are
+		 *   available in 1.8.3 and will be added after lnd no longer
+		 *   supports 1.7, including suites that support CBC mode)
+		 *
+		 * The cipher suites are ordered from strongest to weakest
+		 * primitives, but the client's preference order has more
+		 * effect during negotiation.
+		**/
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+			tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+		},
+		MinVersion: tls.VersionTLS12,
+	}
+	sCreds := credentials.NewTLS(tlsConf)
+	opts := []grpc.ServerOption{grpc.Creds(sCreds)}
+	grpcEndpoint := fmt.Sprintf("localhost:%d", loadedConfig.RPCPort)
+	cCreds, err := credentials.NewClientTLSFromFile(cfg.TLSCertPath,
+		"")
+	if err != nil {
+		return err
+	}
+
+	// In case lnd is started with the --encryptwallet flag, we wait
+	// until the user provide a password over RPC. If this is flag
+	// is not set, we use the default password "hello" for wallet
+	// encryption.
+	walletPw := []byte("hello")
+	if cfg.EncryptWallet {
+		// Set up a new PasswordService, which will listen
+		// for passwords provided over RPC.
+		grpcServer := grpc.NewServer(opts...)
+
+		chainConfig := cfg.Bitcoin
+		if registeredChains.PrimaryChain() == litecoinChain {
+			chainConfig = cfg.Litecoin
+		}
+		pwService := passwordservice.New(chainConfig.ChainDir,
+			activeNetParams.Params)
+		lnrpc.RegisterWalletUnlockerServer(grpcServer, pwService)
+
+		// Start a gRPC server listening for HTTP/2 connections, solely
+		// used for getting the encryption password from the client.
+		lis, err := net.Listen("tcp", grpcEndpoint)
+		if err != nil {
+			fmt.Printf("failed to listen: %v", err)
```

Comment:
Use `rpcsLog` or just return this as `return fmt.Errorf(...)`

Reply:
`fmt.Printf` is used similarly elsewhere in this file, so maybe there's a reason for that? cc @Roasbeef